report/reports.py

```diff
@@ -8,31 +8,63 @@ def count_games(file_name):
 
 def decide(file_name, year):
     with open(file_name) as game_inv:
-        #igy miert fogadja el a test.py? ha kiprintelem nem adja ki a megoldast, de a unittesten atmegy
-        return (True if line.split("\t")[2] == str(year) else False for line in game_inv)
-
-        #igy mukodik, csak hosszu, nem? nem tom h lehetne rovidebben
-        """
-        result = [True if line.split("\t")[2] == str(year) else False for line in game_inv]
-        if True in result:
-            return True
-        else:
-            return False
-        """
+        for line in game_inv:
+            if line.split("\t")[2] == str(year):
+                return True
+        return False
 
 def get_latest(file_name):
+    game = open("game_stat.txt", "r")
+    game_inv = game.readlines()
+    game.close()
+    maximum = max(int(line.split("\t")[2]) for line in game_inv)
+    for line in game_inv:
+        if line.split("\t")[2] == str(maximum):
+            return line.split("\t")[0]
+
+def count_by_genre(file_name, genre):
+    with open(file_name) as game_inv:
+        return sum(1 for x in [line.split("\t") for line in game_inv] if x[3] == genre)
```

Comment:
lehet h igy erthetobb:
`sum(1 for game_genre in [line.split("\t")[3] for line in game_inv] if game_genre == genre)`